Automattic/wp-calypso
# chore: Update Emotion to v11

### Changes proposed in this Pull Request
Update Emotion packages to v11.

The main changes needed were:
1. Rename the packages per https://emotion.sh/docs/emotion-11. (E.g. `@emotion/core` -> `@emotion/react`)
2. Update to the new way of defining themes in Typescript (via a module declaration on `@emotion/react`, and then import `styled` directly from `@emotion/styled`)
3. Manually resolve emotion 11 in storybook. Storybook internally tries to resolve emotion 10 still, which breaks things.
4. Add some return types to appease typescript
5. Remove manual `PropTypes` in some cases. Emotion 11 has better typescript support, so if the `PropTypes` did not exactly match the typescript types, it would fail. So I removed manual `PropTypes` declarations in two places. (They now rely on the Typescript prop types)

### Testing instructions
1. TeamCity should pass, other than some style issues related to use before define (fixed in #55806)
2. `yarn` should succeed locally
3. Storybook should work. In particular, try `composite-checkout:storybook:start` and verify that the components show up in the browser.
4. yarn start/build should work.
5. Smoke test calypso.live
